test/jdk/java/awt/Frame/FrameRepackTest.java

```diff
@@ -0,0 +1,177 @@
+            repacks the frame, thereby adjusting its size tightly to its panel(s).
+
+            Upon test completion, click Pass or Fail appropriately.
+            """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .title("FrameRepackTest Instructions")
+                .instructions(INSTRUCTIONS)
+                .testTimeOut(5)
+                .rows(30)
+                .columns(45)
+                .build();
+
+        SwingUtilities.invokeAndWait(() -> {
+            FrameRepack frame = new FrameRepack();
+
+            PassFailJFrame.addTestWindow(frame);
+            PassFailJFrame.positionTestWindow(frame,
+                    PassFailJFrame.Position.HORIZONTAL);
+
+            frame.setVisible(true);
+        });
+
+        passFailJFrame.awaitAndCheck();
+    }
+
+}
+
+class FrameRepack extends Frame implements ActionListener {
+
+    Panel south;
+
+    public FrameRepack() {
+        super("FrameRepack");
+
+        // create the menubar
+        MenuBar menubar = new MenuBar();
+        this.setMenuBar(menubar);
+        // create the options
+        Menu flip = new Menu("Flip");
+        MenuItem mi;
+        flip.addSeparator();
+        mi = new MenuItem("visible");
+        mi.addActionListener(this);
+        flip.add(mi);
+        mi = new MenuItem("not visible");
+        mi.addActionListener(this);
+        flip.add(mi);
+
+        menubar.add(flip);
+
+        setLayout(new BorderLayout(2, 2));
+        setBackground(Color.green);
+
+        // north panel is always displayed
+        Panel north = new Panel();
+        north.setBackground(Color.red);
+        north.setLayout(new BorderLayout(2, 2));
+        north.add("North", new Label(" This panel is always displayed"));
+        north.add("Center", new Label(" it is a test "));
+        north.setSize(200, 200);
+        add("North", north);
+
+        // south panel can be visible or not...
+        // The problem seems to occur when I put this panel not visible
+        south = new Panel();
+        south.setBackground(Color.white);
+        south.setLayout(new BorderLayout());
+
+        ScrollPane scroller = new ScrollPane(ScrollPane.SCROLLBARS_ALWAYS);
+        scroller.setBackground(Color.yellow);
+        Panel pan1 = new Panel();
+        pan1.setBackground(Color.blue);
+        pan1.setLayout(new BorderLayout());
+
+        pan1.setSize(400, 150);
+        scroller.add("Center", pan1);
+
+        south.add("South", scroller);
+
+        add("South", south);
+
+        south.setVisible(false);
+
+        setSize(350, 300);
+
+        pack();
+    }
+
+    public void actionPerformed(ActionEvent evt) {
+        if (evt.getSource() instanceof MenuItem) {
+            if (evt.getActionCommand().equals("visible")) {
+                south.setVisible(true);
+                pack();
+                setVisible(true);
```

Comment:
Nothing should change if you remove `setVisible(true)` here, the frame is never hidden, is it?